create-rust-app/src/mailer.rs

```diff
@@ -56,13 +89,36 @@ impl Mailer {
         let actually_send: bool = std::env::var("SEND_MAIL")
             .unwrap_or_else(|_| "false".to_string())
             .eq_ignore_ascii_case("true");
+        Mailer {
+            from_address,
+            smtp_server,
+            smtp_username,
+            smtp_password,
+            actually_send,
+        }
+    }
+
+    #[cfg(feature = "plugin_auth")]
+    pub fn new(templates: Box<dyn EmailTemplates>) -> Self {
+        Mailer::check_environment_variables();
 
+        let from_address: String = std::env::var("SMTP_FROM_ADDRESS")
+            .unwrap_or_else(|_| "create-rust-app@localhost".to_string());
+        let smtp_server: String = std::env::var("SMTP_SERVER").unwrap_or_else(|_| "".to_string());
+        let smtp_username: String =
+            std::env::var("SMTP_USERNAME").unwrap_or_else(|_| "".to_string());
+        let smtp_password: String =
+            std::env::var("SMTP_PASSWORD").unwrap_or_else(|_| "".to_string());
+        let actually_send: bool = std::env::var("SEND_MAIL")
+            .unwrap_or_else(|_| "false".to_string())
+            .eq_ignore_ascii_case("true");
         Mailer {
             from_address,
             smtp_server,
             smtp_username,
             smtp_password,
             actually_send,
+            templates,
         }
     }
```

Comment:
expose a different new() function when plugin_auth is enabled that allows the usage of custom templates